tests/api.c

```diff
@@ -3735,6 +3738,231 @@ static int test_wc_Sha224Final (void)
     return 0;
 } /* END test_wc_Sha224Final */
 
+/*
+ * unit test for wc_IdeaSetKey()
+ */
+static int test_wc_IdeaSetKey (void)
+{
+#ifdef HAVE_IDEA
+
+    Idea        idea;
+    const byte  key[] =
+    {
+        0x37, 0x37, 0x37, 0x37, 0x37, 0x37, 0x37, 0x37,
+        0x37, 0x37, 0x37, 0x37, 0x37, 0x37, 0x37, 0x37
+    };
+    int         ret, flag = 0;
+
+    printf(testingFmt, "wc_IdeaSetKey()");
+    /*IV can be NULL, default value is 0*/
+    ret = wc_IdeaSetKey(&idea, key, IDEA_KEY_SIZE, NULL, IDEA_ENCRYPTION);
+    if (ret == 0) {
+        ret = wc_IdeaSetKey(&idea, key, IDEA_KEY_SIZE, NULL, IDEA_DECRYPTION);
+    }
+    /* Bad args. */
+    if (ret == 0) {
+    ret = wc_IdeaSetKey(NULL, key, IDEA_KEY_SIZE, NULL, IDEA_ENCRYPTION);
+        if (ret != BAD_FUNC_ARG) {
+            flag = 1;
+        }
+        ret = wc_IdeaSetKey(&idea, NULL, IDEA_KEY_SIZE, NULL, IDEA_ENCRYPTION);
+        if (ret != BAD_FUNC_ARG) {
+            flag = 1;
+        }
+        ret = wc_IdeaSetKey(&idea, key, IDEA_KEY_SIZE - 1,
+                                    NULL, IDEA_ENCRYPTION);
+        if (ret != BAD_FUNC_ARG) {
+            flag = 1;
+        }
+        ret = wc_IdeaSetKey(&idea, key, IDEA_KEY_SIZE, NULL, -1);
+        if (ret != BAD_FUNC_ARG) {
+            flag = 1;
+        }
+        if (flag == 1) {
+            ret = SSL_FATAL_ERROR;
+        } else {
+            ret = 0;
+        }
+    } /* END Test Bad Args. */
+
+    printf(resultFmt, ret == 0 ? passed : failed);
+
+#endif
+    return 0;
+
+} /* END test_wc_IdeaSetKey */
+
+/*
+ * Unit test for wc_IdeaSetIV()
+ */
+static int test_wc_IdeaSetIV (void)
+{
+#ifdef HAVE_IDEA
+    Idea    idea;
+    int     ret;
+
+    printf(testingFmt, "wc_IdeaSetIV()");
+
+    ret = wc_IdeaSetIV(&idea, NULL);
+    /* Test bad args. */
+    if (ret == 0) {
+        ret = wc_IdeaSetIV(NULL, NULL);
+    }
+    if (ret == BAD_FUNC_ARG) {
```

Comment:
What if the first call to wc_IdeaSetIV() returns with a BAD_FUNC_ARG?  Won't this test still report "passed"?